paymentsheet/src/main/java/com/stripe/android/paymentsheet/elements/common/TextFieldElementState.kt

```diff
@@ -0,0 +1,39 @@
+package com.stripe.android.paymentsheet.elements.common
+
+/**
+ * This represents the different states a field can be in,
+ * in each of these cases there might be a reason to show the
+ * error in a different way
+ */
+internal abstract class TextFieldElementState {
+
+    /**
+     * This indicate the Element state has reached the max number of characters
+     * and focus should shift forward.
+     */
+    abstract fun isFull(): Boolean
+
+    /**
+     * Indicates element is valid and field extraction can happen
+     * and be used to create PaymentMethod Parameters
+     */
+    abstract fun isValid(): Boolean
+
+    /**
+     * If in a state where isValid() returns false it indicates the element cannot be
+     * used for extraction.  This function should return the error message.
+     */
+    abstract fun getErrorMessageResId(): Int?
+
+    open class TextFieldElementStateValid : TextFieldElementState() {
+        override fun isValid(): Boolean = true
+        override fun isFull(): Boolean = false
+        override fun getErrorMessageResId(): Int? = null
+    }
+
+    open class TextFieldElementStateError(private val stringResId: Int) : TextFieldElementState() {
```

Comment:
Do these need to be `open`? Can they be `abstract` or `sealed`?

Reply:
abstract works well here